src/udev/udev-builtin-path_id.c

```diff
@@ -275,7 +281,13 @@ static sd_device *handle_scsi_ata(sd_device *parent, char **path) {
         if (sd_device_get_sysattr_value(atadev, "port_no", &port_no) < 0)
                 return NULL;
 
-        path_prepend(path, "ata-%s", port_no);
+        if (bus != 0)
+                /* Devices behind port multiplier have a bus != 0*/
+                path_prepend(path, "ata-%s.%u.0", port_no, bus);
```

Comment:
Hmm, why is this last number always 0?

Reply:
The number 0 attached in the end is only used to differentiate from the format of master/slave. IMO it doesn't matters too much if it's unique.